wordpress-mobile/WordPress-iOS
# [Insights Management] Add Analytics tracking

Ref #12641 

This is a re-implementation of https://github.com/wordpress-mobile/WordPress-iOS/pull/12695. It was reverted with https://github.com/wordpress-mobile/WordPress-iOS/pull/12715 because Unit Tests were failing.

The problem was `WPAnalyticsStatAutoUploadPostInvoked` was [added](https://github.com/wordpress-mobile/WordPress-iOS-Shared/commit/13ee278c3dc91b8342af3de7f9d126b59709f879#diff-c818c64df9bd03ec0fdacab87d2efb43) to the Shared pod but was not added to WPiOS.

With @yaelirub 's approval, I added a corresponding case to `WPAnalyticsTrackerAutomatticTracks.m`.

To test:

---
- Run the `WPAnalyticsTrackerAutomatticTracks` tests & verify they pass.

---
- On the 'Add New Stats Card' view, select a stat.
- Verify `stats_add_insight_item_selected <insight: Insight Title>` is logged.

---
- On a stat header, tap the Manage Insights button.
  - Verify `stats_manage_insight_tapped` is logged.
- Select 'Move up'.
  - Verify `stats_insight_move_up_tapped` is logged.
- Select 'Move down'.
  - Verify `stats_insight_move_down_tapped` is logged.
- Select 'Remove from Insights'
  - Verify `stats_insight_remove_tapped <insight: Insight Title>` is logged.

---
- Remove all stats from Insights.
- Tap the 'Add stats card' button.
- Verify `stats_add_insight_item_tapped` is logged.

---
Update release notes:
